test/integration/integration.cc

```diff
@@ -545,12 +545,25 @@ void BaseIntegrationTest::useListenerAccessLog(absl::string_view format) {
   ASSERT_TRUE(config_helper_.setListenerAccessLog(listener_access_log_name_, format));
 }
 
-std::string BaseIntegrationTest::waitForAccessLog(const std::string& filename) {
+size_t entryIndex(const std::string& file, uint32_t entry) {
+  size_t index = 0;
+  for (uint32_t i = 0; i < entry; ++i) {
+    index = file.find('\n', index);
+    if (index == std::string::npos || index == file.length()) {
+      return file.length();
+    }
+    ++index;
+  }
+  return index;
+}
+
+std::string BaseIntegrationTest::waitForAccessLog(const std::string& filename, uint32_t entry) {
```

Comment:
this whole function looks like potential source of flakiness, though that doesn't change with this PR.

I really want to have a memory-file system where we can set triggers on any write to do the tests, without having to use real-time sleeps and limits. That's not really an issue for this PR of course but interested to hear your opinions about this.

Reply:
Agreed - I was actually thinking of doing that as part of the PR, but decided I'd rather not block on it.
It's on my "non-essential but worthwhile cleanups" list because I was pretty bummed out at it not existing, and then got annoyed again when the existing "wait for logs" function didn't work because 1) it didn't handle multiple logs and 2) it wasn't integrated with simtime (sigh)

Reply:
maybe in the context of this PR should we consider changing some time-constants to make flakiness less likely, like give it 10 or 30 seconds to run and wake-up only 10 to 30ms?

Reply:
This is using your simtime.   We advance time to the point the logs should be flushed, and then the 'wait for logs" function loops to give the server thread time to realize the flush should happen.  I don't think it's likely to flake, and I don't think changing the time constants would help any?

Reply:
Per my other comment above, advanceTimeAsync does not block; sorry for suggesting it. I think this will be more robust if you switch to advanceTimeWait, assuming the log is actually written on a call-stack triggered by a timer.